include/s3select_functions.h

```diff
@@ -1231,69 +1236,728 @@ struct _fn_in : public base_function
+        v.push_back('.');
+        v.push_back('*');
+        v.push_back('$');
+        st = END;
+        } else if(*p == '%') {
+        v.push_back('.');
+        v.push_back('*');
+        st = STAR_CHAR;
+        } else  if(last_char(*p)) {
+        v.push_back(*p);
+        v.push_back('$');
+        st = END;
+        } else {
+        v.push_back(*p);
+        st = ANYCHAR;
+        } break;
+
+      case METACHAR:
+          if(*p == escape) {
+          st = ESCAPE;
+          } else if(*p == '_' && last_char(*p)) {
+        v.push_back('.');
+        v.push_back('$');
+        st = END;
+        } else if(*p == '_') {
+        v.push_back('.');
+        st = METACHAR;
+        } else if(*p == '%' && last_char(*p)) {
+        v.push_back('.');
+        v.push_back('*');
+        v.push_back('$');
+        st = END;
+        } else if(*p == '%') {
+        v.push_back('.');
+        v.push_back('*');
+        st = STAR_CHAR;
+        } else if(last_char(*p)) {
+        v.push_back(*p);
+        v.push_back('$');
+        st = END;
+        } else {
+        v.push_back(*p);
+        st = ANYCHAR;
+        } break;
+
+      case ANYCHAR:
+          if(*p == escape) {
+          st = ESCAPE;
+          } else if(*p == '_' && last_char(*p)) {
+        v.push_back('.');
+        v.push_back('$');
+        st = END;
+        } else if(*p == '_') {
+        v.push_back('.');
+        st = METACHAR;
+        } else if(*p == '%' && last_char(*p)) {
+        v.push_back('.');
+        v.push_back('*');
+        v.push_back('$');
+        st = END;
+        } else if(*p == '%') {
+        v.push_back('.');
+        v.push_back('*');
+        st = STAR_CHAR;
+        } else  if(last_char(*p)) {
+        v.push_back(*p);
+        v.push_back('$');
+        st = END;
+        } else {
+        v.push_back(*p);
+        st = ANYCHAR;
+        } break;
+
+      case STAR_CHAR:
+          if(*p == escape) {
+          st = ESCAPE;
+          } else if(*p == '%' && last_char(*p)) {
+        v.push_back('$');
+        st = END;
+        } else if(*p == '%') {
+        st = STAR_CHAR;
+        } else if(*p == '_' && last_char(*p)) {
+        v.push_back('.');
+        v.push_back('$');
+        st = END;
+        } else if(*p == '_') {
+        v.push_back('.');
+        st = METACHAR;
+        } else  if(last_char(*p)) {
+        v.push_back(*p);
+        v.push_back('$');
+        st = END;
+        } else {
+        v.push_back(*p);
+        st = ANYCHAR;
+        } break;
+
+      case ESCAPE:
+        if(last_char(*p)) {
+        v.push_back(*p);
+        v.push_back('$');
+        st = END;
+        } else {
+        v.push_back(*p);
+        st = ANYCHAR;
+        } break;
+
+      case END:
+        return v;
+
+      default:
+        throw base_s3select_exception("missing state!");
+        break;
+    }
+    p++;
+  }
+  return v;
+}
+
+bool operator()(bs_stmt_vec_t* args, variable* result) override
+  {
+    auto iter = args->begin();
```

Comment:
the operator should check if it need run ->eval() on like-expr, escape-expr.
in case it need to run ->eval(), then it should run also transform and regex-compile.